_posts/2017-02-22-ChainerRL-Deep-Reinforcement-Learning-Library.md

```diff
@@ -0,0 +1,114 @@
+layout: post
+categories: General
+---
+
+Chainer-based deep reinforcement learning library, ChainerRL has been released.
+[https://github.com/pfnet/chainerrl](https://github.com/pfnet/chainerrl)
+
+(This post is translated from the [original post](https://research.preferred.jp/2017/02/chainerrl/) written by Yasuhiro Fujita.)
+
+### Algorithms
+
+ChainerRL contains a set of Chainer implementations of deep reinforcement learning (DRL) algorithms. The followings are implemented and accessible under a unified interface.
+
+* Deep Q-Network [(Mnih et al., 2015)](http://www.nature.com/nature/journal/v518/n7540/full/nature14236.html)
+* Double DQN [(Hasselt et al., 2016)](https://arxiv.org/abs/1509.06461)
+* Normalized Advantage Function [(Gu et al., 2016)](https://arxiv.org/abs/1603.00748)
+* (Persistent) Advantage Learning [(Bellemare et al., 2016)](http://arxiv.org/abs/1512.04860)
+* Deep Deterministic Policy Gradient (DDPG) [(Lillicrap et al., 2016)](http://arxiv.org/abs/1509.02971)
+* SVG(0) [(Heese et al., 2015)](https://arxiv.org/abs/1510.09142)
+* Asynchronous Advantage Actor-Critic (A3C) [(Mnih et al., 2016)](http://arxiv.org/abs/1602.01783)
+* Asynchronous N-step Q-learning [(Mnih et al., 2016)](http://arxiv.org/abs/1602.01783)
+* Actor-Critic with Experience Replay [(Wang et al., 2017)](http://arxiv.org/abs/1611.01224)
+* etc.
+
+### Examples
+
+The ChainerRL library comes with many examples such as video gameplay of Atari 2600 using A3C,
+
+![Atari 2600 play]({{ site.url }}/assets/chainerrl_breakout.gif)
+
+and learning to control humanoid robot using DDPG.
+
+![Humanoid]({{ site.url }}/assets/chainerrl_humanoid.gif)
+
+### How to use
+
+Here is a brief introduction to ChainerRL.
+
+First, user must provide an appropriate definition of the problem (called "environment") that is to be solved using reinforcement learning. The format of defining the environment in ChainerRL follows that of OpenAI's Gym ([https://github.com/openai/gym](https://github.com/openai/gym)), a benchmark toolkit for reinforcement learning. ChainerRL can be used either with Gym or an original implementation of environment. Basically, the environment should have two methods, `reset()` and `step()`.
+
+{% highlight python linenos %}
+env = YourEnv()
+# reset() returns the current observation given the envirnment
+obs = env.reset()
+action = 0
+# step() sends an action to the environemnt, then returns four values (next observation, reward, whether it reachs the terminal of episode, and additional information).
+obs, r, done, info = env.step(action)
+{% endhighlight %}
+
+In DRL, neural networks correspond to policy and value functions (V-function or Q-function), which determines action given state, and estimates value of state or action, respectively. The parameters of neural network models are then updated through training. In ChainerRL, policies and value functions are represented as `Link` of Chainer, which just implements `__call__()` method.
```

Comment:
I thought we can interpret the sentence as `Link` being an object that implements `__call__()`. But in general, `Link` need not to implement `__call__` method. So how about writing it as "In Chainer RL policies and value functions are represented as a `Link` object in Chainer that implements `__call__()` method" ?